### PR TITLE
Add observedGeneration to status of alert and receivers

### PR DESCRIPTION
**Pull request by SomtochiAma**
Signed-off-by: Somtochi Onyekwere <somtochionyekwere@gmail.com>

### CODE REVIEW

**Comment by stefanprodan** [comment]
@SomtochiAma we need to also set it like this: https://github.com/fluxcd/kustomize-controller/blob/main/api/v1beta1/kustomization_types.go#L251

File: controllers/alert_controller.go [hunk 1]
```diff
@@ -88,6 +88,8 @@ func (r *AlertReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl
 
 	r.recordReadiness(ctx, alert)
 
+	alert.Status.ObservedGeneration = alert.Generation
```
Comment (stefanprodan) [inline controllers/alert_controller.go:91]: Hmm what is this supposed to do?

Comment (SomtochiAma) [reply controllers/alert_controller.go:91]: Setting it like you stated above?

Comment (stefanprodan) [reply controllers/alert_controller.go:91]: What about persistence?

**Review by stefanprodan — APPROVED** [review]
LGTM

Thanks @SomtochiAma

**Pull request merged**